### PR TITLE
feat(assets): add token details screen

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

[Figma](https://www.figma.com/file/OE6RFzqXAMqU9ygHtNYyof/Assets-Page-(Multichain)?node-id=2099%3A15029&mode=dev)

### Test plan

Unit tests, manual by enabling dummy feature flag

Showing all actions with swap enabled:

https://github.com/valora-inc/wallet/assets/5062591/5041860d-d332-49e2-97cc-8abcaf214e7c

Showing actions with swap disabled (withdraw is hidden behind more):

https://github.com/valora-inc/wallet/assets/5062591/0d84a6ad-bde8-4a41-8b25-4aa67dac950e

### Related issues

- Fixes ACT-915

### Backwards compatibility

Yes

### CODE REVIEW

**Comment by cajubelt** [comment]
just from the videos-
- the bottom sheet when "more" is clicked isn't in the demo video
- the "withdraw" icon looks kinda close to the edge. it's not in the designs, so this is pretty subjective

very cool though!

File: src/tokens/hooks.ts
Comment (cajubelt) [inline src/tokens/hooks.ts:71]: is there an issue where this will be done? or is it something you plan to do in a followup PR?

File: src/tokens/hooks.ts
Comment (cajubelt) [inline src/tokens/hooks.ts:79]: same here

File: src/tokens/hooks.ts
Comment (cajubelt) [inline src/tokens/hooks.ts:92]: what do you think about abandoning the "CiCo" casing pattern? seems like proper camelCase would just be CICO

File: src/tokens/hooks.ts
Comment (cajubelt) [inline src/tokens/hooks.ts:96]: see in cases like this one we capitalize the i, just not when we abbreviate. doesn't make sense to me

**Comment by MuckT** [comment]
@satish-ravi I got an app crash when navigating around the tabs

File: src/tokens/TokenDetails.tsx
Comment (cajubelt) [inline src/tokens/TokenDetails.tsx:143]: I recently saw a refactor [here](https://github.com/valora-inc/wallet/pull/4242) by @MuckT that changes the meaning of defaultTokenOverride from address to tokenId, but the type is still string. Some faint alarm bell went off when I saw that, and I'm realizing now that I was worried that one of you could merge the changes without seeing this change and we'd end up with a bug. @MuckT what do you think should be done to prevent this? Maybe the `defaultTokenOverride` parameter name could be changed (in Tom's PR) so we get a compilation error if someone makes this mistake

File: src/tokens/TokenDetails.tsx
Comment (cajubelt) [inline src/tokens/TokenDetails.tsx:165]: this type cast seems to rely on the internal logic of `useCashInTokens`, which could change and cause a bug here...

Comment (cajubelt) [reply src/tokens/TokenDetails.tsx:165]: same with the following one

File: src/tokens/TokenDetails.tsx
Comment (cajubelt) [inline src/tokens/TokenDetails.tsx:196]: nit: magic numbers. probly better to put 3 into `MAX_BUTTONS_PER_ROW`

File: src/tokens/TokenDetails.tsx
Comment (cajubelt) [inline src/tokens/TokenDetails.tsx:202]: magic numbers

**Review by cajubelt — APPROVED** [review]
main concerns are a type cast and a potential conflict with Tom's PR-- can you work with him to resolve it?

nothing major so I'm approving also

**Comment by satish-ravi** [comment]
@cajubelt
- the more bottom sheet is out of scope for this and will be covered in a separate ticket. 
- I've been chatting w/ Nitya on the Withdraw button, I think she's ok with this since for most users swap would be enabled and withdraw will only pop up on the bottom sheet. This would also be a problem for all buttons in other languages if there are longer names, but I think we can hold off on optimizing it for all languages now.

File: src/tokens/TokenDetails.tsx
Comment (MuckT) [reply src/tokens/TokenDetails.tsx:143]: Happy to change it from `defaultTokenOverride` to `defaultTokenIdOverride` in #4242.

File: src/tokens/TokenDetails.tsx
Comment (satish-ravi) [reply src/tokens/TokenDetails.tsx]: fixed

File: src/tokens/TokenDetails.tsx
Comment (satish-ravi) [reply src/tokens/TokenDetails.tsx]: fixed

File: src/tokens/TokenDetails.tsx
Comment (satish-ravi) [reply src/tokens/TokenDetails.tsx]: added an explicit condition on the navigate function

File: src/tokens/hooks.ts
Comment (satish-ravi) [reply src/tokens/hooks.ts]: todo addressed after #4265 was merged

File: src/tokens/hooks.ts
Comment (satish-ravi) [reply src/tokens/hooks.ts]: I ended up changing to Cico to be consistent with the feature gate `showCico`

File: src/tokens/TokenDetails.tsx
Comment (satish-ravi) [reply src/tokens/TokenDetails.tsx:159]: Left a TODO, so this can be addressed depending on which one merges first

Comment (cajubelt) [reply src/tokens/TokenDetails.tsx:159]: I think Tom's suggestion of changing the variable name is safer, since it doesn't depend on merge order-- in either case, a build error should take place if one is merged without the other taking the changes into account yet

**Review by cajubelt — APPROVED** [review]

**Pull request merged**
